app/javascript/components/upload/uploadUtils.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faDna } from '@fortawesome/free-solid-svg-icons'
+
+/** properties used to track file state on the form, but that should not be sent to the server
+ *  this also includes properties that are only modifiable on the server (and so should also
+ * be ignored server side, but for best hygiene are also just not sent ) */
+const PROPERTIES_NOT_TO_SEND = [
+  'selectedFile',
+  'uploadSelection',
+  'submitData',
+  'isDirty',
+  'isSaving',
+  'isError',
+  'generation',
+  'created_at',
+  'updated_at',
+  'queued_for_deletion',
+  'data_dir',
+  'options',
+  'version',
+  'status',
+  'upload',
+  'parse_status'
+]
+
+/** reworks the file object to make it easier to work with
+  * maps the id property to _id, and sets undefined properties
+  * this modifies the object in-place, but also returns it for easy chaining
+  */
+export function formatFileFromServer(file) {
+  file._id = file._id.$oid
+  file.description = file.description ? file.description : ''
+  file.study_id = file.study_id.$oid
```

Comment:
This would make it more JS-idiomatic, and also warn developers that they're dealing with convenient shorthand that can't be used for e.g. posting upstream.

```suggestion
  file.studyId = file.study_id.$oid
```

Reply:
actually, this is used for posting upstream to `study_files_controller.rb`, which is why it is camel-cased